src/limbo/bayes_opt/boptimizer.hpp

```diff
@@ -113,6 +114,9 @@ namespace limbo {
                     } else {
                         _model.add_sample(this->_samples.back(), this->_observations.back(), Params::bayes_opt_boptimizer::noise());
                     }
+                    if (this->_current_iteration % Params::bayes_opt_boptimizer::hp_period() == 0
+                        || Params::bayes_opt_boptimizer::hp_period() == -1)
```

Comment:
@jbmouret What's the use of the second check (`hp_period()==-1`), since **x%1** is **always equal to 0**? So, if someone wants to optimize in every iteration, he should set the `hp_period` to 1.

We should use `hp_period() == -1` to not optimize at all. What do you think?